engineapi/proxy_backup.go

```diff
@@ -51,3 +54,28 @@ func (p *Proxy) SnapshotBackupStatus(e *longhorn.Engine, backupName, replicaAddr
 
 	return (*longhorn.EngineBackupStatus)(recv), nil
 }
+
+func (p *Proxy) BackupRestore(e *longhorn.Engine, backupTarget, backupName, backupVolumeName, lastRestored string, credential map[string]string) error {
+	backupURL := backupstore.EncodeBackupURL(backupName, backupVolumeName, backupTarget)
+
+	// get environment variables if backup for s3
+	envs, err := getBackupCredentialEnv(backupTarget, credential)
+	if err != nil {
+		return err
+	}
+
+	taskError, err := p.grpcClient.BackupRestore(p.DirectToURL(e), backupURL, backupTarget, backupVolumeName, envs)
```

Comment:
NIT: Can the gRPC Client aggregate the errors into a single one?

Reply:
Addressed.